### ISSUE TITLE
emacs-eclim broken on the stable version?

### ISSUE THREAD

**Issue by emmanueltouzery** [syl20bnr/spacemacs#6949]
#### Description
emacs-eclim doesn't work for me anymore with the latest stable release (0.105.22). It worked with 0.105.21 and the old eclim package.

#### Reproduction guide
- Start Emacs with the java layer enabled
- Open a java file

*Observed behaviour:*
Getting messages:

    Company backend ’company-emacs-eclim’ could not be initialized:
    Symbol’s function definition is void: company-emacs-eclim
    eclim--problems-mode-init: Symbol’s value as variable is void: G374

*Expected behaviour:*
The java file should open without warnings and if the eclim server is running, completion, go to definition, etc should work.

#### System Info
- OS: gnu/linux
- Emacs: 25.1.1
- Spacemacs: 0.105.22
- Spacemacs branch: master (rev. 9f9faa4)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(auto-completion :variables auto-completion-enable-help-tooltip t)
```
The `company-emacs-eclim` dependency does not appear to get installed, but even if it did, I think it would not work. I tried to force install it manually, to no avail:

    Contacting host: melpa.org:443
    Making version-control local to company-emacs-eclim-autoloads.el while let-bound!
    Generating autoloads for company-emacs-eclim.el...done
    Wrote /home/emmanuel/.emacs.d/elpa/company-emacs-eclim-20160813.1754/company-emacs-    eclim-autoloads.el
    Checking /home/emmanuel/.emacs.d/elpa/company-emacs-eclim-20160813.1754... [3 times]
    Compiling /home/emmanuel/.emacs.d/elpa/company-emacs-eclim-20160813.1754/company-emacs-eclim.el...done
    Wrote /home/emmanuel/.emacs.d/elpa/company-emacs-eclim-20160813.1754/company-emacs-eclim.elc
    Checking /home/emmanuel/.emacs.d/elpa/company-emacs-eclim-20160813.1754...
    Done (Total of 1 file compiled, 2 skipped)
    Package menu: Operation finished [Installed 1]
    ReportingService.java has auto save data; consider M-x recover-this-file
    eclim--problems-mode-init: Symbol’s value as variable is void: G374

**Comment by TheBB** [comment]
Yeah looks like more changed than just the name. company-emacs-eclim is now a separate package. Can someone who actually uses the java layer fix this? I don't have time to set up eclimd myself to know when it's working.

**Comment by TheBB** [comment]
Here is at least a trivial attempt. I haven't verified it though. It'd be good if someone could try it.

https://github.com/syl20bnr/spacemacs/pull/6954

**Comment by TheBB** [comment]
The above PR has been merged. There are still issues with eclimd which have been reported upstream. See https://github.com/emacs-eclim/emacs-eclim/issues/7. Keeping this open for now.

**Comment by TheBB** [comment]
I see upstream has been closed, Can anyone confirm it works?

**Comment by mattjbray** [comment]
Updated to `eclim-20160901.16` and everything is working well for me 👍

**Comment by TheBB** [comment]
Okay!

**Comment by zarzen** [comment]
using spacemacs stable version?
I have checked the eclim version is `eclim-20160901.16`, but I still have problem 
```
spacemacs/java-completing-dot: Symbol's function definition is void: company-emacs-eclim
```
@mattjbray

**Comment by mattjbray** [comment]
@zarzen I tested using the develop branch. You need #6954 at least.

**Comment by emmanueltouzery** [comment]
could the fix be backported to the stable version? No time to test the patch these days myself, but @mattjbray  confirmed it's ok on develop now.

**Comment by TheBB** [comment]
There will be a new release soon.

**Comment by TheBB** [comment]
Reopening since this isn't fixed on master.

**Comment by zarzen** [comment]
@mattjbray Thanks

**Comment by rubin55** [comment]
Any news on the new stable release or is there a way I can backport this fix into my own stable setup easily myself?

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
